index.bs

```diff
@@ -1046,101 +1013,483 @@ BigIntValue = {
+   of steps for which the associated condition is true, if any:
+
+  <dl>
+    <dt>[=Type=](|value|) is undefined
+    <dd>Let |remote value| be a map matching the <code>UndefinedValue</code>
+    production in the [=local end definition=].
+
+    <dt>[=Type=](|value|) is Null
+    <dd>Let |remote value| be a map matching the <code>NullValue</code>
+    production in the [=local end definition=].
+
+    <dt>[=Type=](|value|) is String
+    <dd>Let |remote value| be a map matching the <code>StringValue</code>
+    production in the [=local end definition=], with the <code>value</code>
+    property set to |value|.
+
+    Issue: This doesn't handle lone surrogates
+
+    <dt>[=Type=](|value|) is Number
+    <dd>
+    1. Switch on the value of |value|:
+      <dl>
+        <dt>NaN
+        <dd>Let |serialized| be <code>"NaN"</code>
+        <dt>-0
+        <dd>Let |serialized| be <code>"-0"</code>
+        <dt>+Infinity
+        <dd>Let |serialized| be <code>"+Infinity"</code>
+        <dt>-Infinity
+        <dd>Let |serialized| be <code>"-Infinity"</code>
+        <dt>Otherwise:
+        <dd>Let |serialized| be |value|
+      </dl>
+
+    1. Let |remote value| be a map matching the <code>NumberValue</code>
+       production in the [=local end definition=], with the <code>value</code>
+       property set to |serialized|.
+
+    <dt>[=Type=](|value|) is Boolean
+    <dd>Let |remote value| be a map matching the <code>BooleanValue</code>
+        production in the [=local end definition=], with the <code>value</code>
+        property set to |value|.
+
+    <dt>[=Type=](|value|) is BigInt
+    <dd>Let |remote value| be a map matching the <code>BigIntValue</code>
+        production in the [=local end definition=], with the <code>value</code>
+        property set to the result of running the [=ToString=] operation on
+        |value|.
+
+  </dl>
+
+1. Return |remote value|
+
+</div>
+
+#### Deserialization #### {#data-types-protocolValue-primitiveProtocolValue-deserialization}
+
+<div algorithm>
+
+To <dfn>deserialize primitive protocol value</dfn> given a |primitive protocol value|:
+
+1. Let |type| be the value of the <code>type</code> field of |primitive protocol value|.
+
+1. Let |value| be undefined.
+
+1. If |primitive protocol value| has field <code>value</code>:
+   1. Let |value| be the value of the <code>value</code> field of |primitive protocol value|.
+
+1. In the following list of conditions and associated steps, run the first set of steps for which
+   the associated condition is true:
+
+  <dl>
+
+    <dt>|type| is the string "<code>undefined</code>"
+    <dd>Return [=success=] with data [=undefined=].
+
+    <dt>|type| is the string "<code>null</code>"
+    <dd>Return [=success=] with data [=null=].
+
+    <dt>|type| is the string "<code>string</code>"
+    <dd>Return [=success=] with data |value|.
+
+    <dt>|type| is the string "<code>number</code>"
+    <dd>Return [=success=] with data [=Number=](|value|).
```

Comment:
Similarly to the BigInt case I think it's clearer to use https://tc39.es/ecma262/#sec-stringtonumber directly.

Reply:
done